nemo/collections/tts/modules/submodules.py

```diff
@@ -732,9 +743,16 @@ def output_types(self):
             "embs": NeuralType(('B', 'D'), EncodedRepresentation()),
         }
 
-    def forward(self, speaker=None, reference_spec=None, reference_spec_lens=None):
+    def overwrite_precomputed_emb(self, emb):
+        self.precomputed_emb = torch.nn.Parameter(emb)
+
+    def forward(self, batch_size, speaker=None, reference_spec=None, reference_spec_lens=None):
```

Comment:
Could default batch_size to None since only the precomputed branch uses it.

Reply:
Done!